File: examples/src/main/java/org/ergoplatform/example/ErgoTool.js
const Long = Java.type("java.lang.Long")
const RestApiErgoClient = Java.type("org.ergoplatform.example.util.RestApiErgoClient")
const ErgoClientException = Java.type("org.ergoplatform.appkit.ErgoClientException")
const ConstantsBuilder = Java.type("org.ergoplatform.appkit.ConstantsBuilder")
const ErgoContract = Java.type("org.ergoplatform.appkit.ErgoContract")
const ErgoToolConfig = Java.type("org.ergoplatform.ergotool.ErgoToolConfig")
const Parameters = Java.type("org.ergoplatform.appkit.Parameters")

const amountToPay = Long.parseLong(process.argv[2]);
const conf = ErgoToolConfig.load("ergotool.json");
const nodeConf = conf.getNode();
const ergoClient = RestApiErgoClient.create(
    nodeConf.getNodeApi().getApiUrl(),
    nodeConf.getNetworkType(),
    nodeConf.getNodeApi().getApiKey());

const newBoxDelay = 30;

const txJson = ergoClient.execute(function (ctx) {
    const wallet = ctx.getWallet();
    const totalToSpend = amountToPay + Parameters.MinFee;
    const boxes = wallet.getUnspentBoxes(totalToSpend);
    if (!boxes.isPresent())
        throw new ErgoClientException("Not enough coins in the wallet to pay " + totalToSpend, null);

    const prover = ctx.newProverBuilder()
        .withMnemonic(
            nodeConf.getWallet().getMnemonic(),
            nodeConf.getWallet().getPassword())
        .build();

    const txB = ctx.newTxBuilder();
    const newBox = txB.outBoxBuilder()
        .value(amountToPay)
        .contract(ctx.compileContract(
            ConstantsBuilder.create()
                .item("deadline", ctx.getHeight() + newBoxDelay)
                .item("pkOwner", prover.getP2PKAddress().pubkey())
                .build(),
            "{ sigmaProp(HEIGHT > deadline) && pkOwner }"))
        .build();
    const tx = txB.boxesToSpend(boxes.get())
        .outputs(newBox)
        .fee(Parameters.MinFee)
        .sendChangeTo(prover.getP2PKAddress())
        .build();

    const signed = prover.sign(tx);
    const txId = ctx.sendTransaction(signed);
    return signed.toJson(true);
});

print(txJson);

File: examples/src/main/java/org/ergoplatform/example/ErgoTool.py
import sys
import java

Long = java.type("java.lang.Long")
RestApiErgoClient = java.type("org.ergoplatform.example.util.RestApiErgoClient")
ErgoClientException = java.type("org.ergoplatform.appkit.ErgoClientException")
ConstantsBuilder = java.type("org.ergoplatform.appkit.ConstantsBuilder")
ErgoContract = java.type("org.ergoplatform.appkit.ErgoContract")
ErgoToolConfig = java.type("org.ergoplatform.ergotool.ErgoToolConfig")
Parameters = java.type("org.ergoplatform.appkit.Parameters")

amountToPay = Long.parseLong(sys.argv[1])
conf = ErgoToolConfig.load("ergotool.json")
nodeConf = conf.getNode()
ergoClient = RestApiErgoClient.create(
    nodeConf.getNodeApi().getApiUrl(),
    nodeConf.getNetworkType(),
    nodeConf.getNodeApi().getApiKey())

newBoxDelay = 30

def send(ctx):
    wallet = ctx.getWallet()
    totalToSpend = amountToPay + Parameters.MinFee
    boxes = wallet.getUnspentBoxes(totalToSpend)
    if not boxes.isPresent():
        raise ErgoClientException("Not enough coins in the wallet to pay " + totalToSpend, None);

    prover = ctx.\
        newProverBuilder().\
        withMnemonic(
            nodeConf.getWallet().getMnemonic(),
            nodeConf.getWallet().getPassword()).\
        build()

    txB = ctx.newTxBuilder()

    newBox = txB.outBoxBuilder().\
        value(amountToPay).\
        contract(ctx.compileContract(
            ConstantsBuilder.create()
                .item("deadline", ctx.getHeight() + newBoxDelay)
                .item("pkOwner", prover.getP2PKAddress().pubkey()).build(),
            "{ sigmaProp(HEIGHT > deadline) && pkOwner }")).\
            build()

    tx = txB.boxesToSpend(boxes.get()).\
        outputs(newBox).\
        fee(Parameters.MinFee).\
        sendChangeTo(prover.getP2PKAddress()).\
        build();

    signed = prover.sign(tx)
    txId = ctx.sendTransaction(signed)
    return signed.toJson(True)

txJson = ergoClient.execute(send)

print(txJson)
